instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/NetClientAttributesGetter.java

```diff
@@ -20,27 +20,21 @@
   @Nullable
   String getTransport(REQUEST request, @Nullable RESPONSE response);
 
-  // TODO: make required after the 1.24 release
   /**
    * Returns the application protocol used.
    *
    * <p>Examples: `amqp`, `http`, `mqtt`.
    */
   @Nullable
-  default String getProtocolName(REQUEST request, @Nullable RESPONSE response) {
-    return null;
-  }
+  String getProtocolName(REQUEST request, @Nullable RESPONSE response);
 
-  // TODO: make required after the 1.24 release
   /**
    * Returns the version of the application protocol used.
    *
    * <p>Examples: `3.1.1`.
    */
   @Nullable
-  default String getProtocolVersion(REQUEST request, @Nullable RESPONSE response) {
-    return null;
-  }
+  String getProtocolVersion(REQUEST request, @Nullable RESPONSE response);
```

Comment:
@lmolkova do you have any thought how common the protocol name/version will be, and whether we should have default methods for these (similar to the sock attributes below), or whether it's good to force implementers to think about / implement these?

Reply:
BTW sort of related: I've been thinking of making `getTransport()` optional, since there are few instrumentations that actually know the transport protocol (instead of just hardcoding `TCP_IP` and hoping it's correct)

Reply:
it makes sense for protocol information to be left unset by default - it does not seem that essential and instrumentations that care can override. 

wrt transport - I think we removed it from HTTP spec completely. HTTP, gRPC, and most of the other protocols would always have TCP there, maybe we should just add a default value to the spec?

Reply:
[HTTP/3](https://en.wikipedia.org/wiki/HTTP/3) uses UDP. gRPC offers an in-process transport for which the hardcoded `IP_TCP` is probably wrong.

Reply:
I'll make the transport optional in a separate PR -- and the protocol attributes in this PR, since it kinda reverts some of my changes.

Reply:
> [HTTP/3](https://en.wikipedia.org/wiki/HTTP/3) uses UDP. gRPC offers an in-process transport for which the hardcoded `IP_TCP` is probably wrong.

Agree, I think the default on the instrumentation side should be `null`. Spec can have language (for the backends) to assume transport from protocol name/version. E.g. HTTP/3 is UDP, there is no need to duplicate it on telemetry.

For less popular protocols, it might make sense to set both protocol name/version AND transport, but their specs can require transport if needed.